lib/r10k/environment/base.rb

```diff
@@ -57,6 +64,22 @@ def initialize(name, basedir, dirname, options = {})
                                          force: @overrides.dig(:modules, :force),
                                          puppetfile_name: @puppetfile_name})
     @puppetfile.environment = self
+
+    loader_options = { basedir: @full_path, overrides: @overrides, environment: self }
+    loader_options[:puppetfile] = @puppetfile_name if @puppetfile_name
+
+    if @overrides.dig(:environments, :assume_unchanged)
+      @pre_loader = R10K::ModuleLoader::StubPuppetfile.new(**loader_options)
+      existing_modules = @pre_loader.load[:modules].map {|mod| [ mod.name, mod.version ] }.to_h
```

Comment:
this is pretty dense line compared to most of this code base. the formatting is also my own odd style.

Reply:
I don't mind it 🤷‍♀️